File: C2/PDOO/Practicas/DeepSpaceRuby/lib/SpaceStationToUI.rb

module Deepspace


# 5.4.17 - Translation from Java
# @author Profesor
  
class SpaceStationToUI
  
  attr_reader :name, :nMedals, :ammoPower, :fuelUnits, \
      :shieldPower, :shieldBoosters, :hangar, :weapons, \
      :pendingDamage
  
  def initialize (station)
    @weapons=Array.new()
    for w in station.weapons do
      @weapons << w.getUIversion()
    end
    
    @shieldBoosters=Array.new() 
    for sb in station.shieldBoosters do
      @shieldBoosters << sb.getUIversion()
    end
    
    @name=station.name
    @nMedals=station.nMedals
    
    @ammoPower=station.ammoPower
    @fuelUnits=station.fuelUnits
    @shieldPower=station.shieldPower
    
    h=station.hangar()
    if (h!=nil) then
      @hangar=h.getUIversion()
    else 
      @hangar=nil
    end
    
    #MIGUEL, en Java el pendingDamage no se incluye en la versión ToUI
    pd = station.pendingDamage
    if (pd != nil) then 
      @pendingDamage=pd.getUIversion()
    else
      @pendingDamage=nil
    end
  end
  
  def to_s
    out="Space Station - Name: #{name}\n"
    out+="\tnMedals: #{@nMedals}, Fuel units: #{@fuelUnits.round(2)}, Power: #{@ammoPower}, Shields: #{@shieldPower}\n"
    out+="\tWeapons: [#{@weapons.join(', ')}]\n"
    out+="\tShieldBooster: [#{@shieldBoosters.join(', ')}]\n"
    out+="\tHangars: #{@hangar}\n"
    out+="\tPendingDamage: #{@pendingDamage}\n" 
    out+="------- end of Space Station >> #{@name} << -------"
    return out
  end

end # class

end # module

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/SpaceStationToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

import java.util.ArrayList;

/**
 *
 * @author Profesor
 */
public class SpaceStationToUI {
    private String name;
    private int nMedals;
    
    private float ammoPower=1.0f;
    private float fuelUnits=1.0f;
    private float shieldPower=1.0f;
    
    private ArrayList<WeaponToUI> weapons;
    private ArrayList<ShieldToUI> shieldBoosters;
    private HangarToUI hangar;   
    private DamageToUI pendingDamage;
    
    SpaceStationToUI(SpaceStation station) {
        weapons=new ArrayList();
        shieldBoosters=new ArrayList();         
        
        name=station.getName();
        nMedals=station.getNMedals();
        
        ammoPower=station.getAmmoPower();
        fuelUnits=station.getFuelUnits();
        shieldPower=station.getShieldPower();
        
        for (Weapon w:station.getWeapons()) {
            weapons.add(w.getUIversion());
        }
        
        for(ShieldBooster s:station.getShieldBoosters()) {
            shieldBoosters.add(s.getUIversion());
        }
        
        Hangar h=station.getHangar();
        if (h!=null) {
            hangar=h.getUIversion();
        }
        else {
            hangar=null;
        }
        
        // MIGUEL: Añadido de  pendingDamage  y su consultor
        
        Damage d = station.getPendingDamage();
        if (!d.hasNoEffect()) {
          pendingDamage = d.getUIversion();
        } else {
          pendingDamage = null;
        }
            
    }

    public String getName() {
        return name;
    }

    public int getnMedals() {
        return nMedals;
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public float getFuelUnits() {
        return fuelUnits;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public ArrayList<WeaponToUI> getWeapons() {
        return weapons;
    }

    public ArrayList<ShieldToUI> getShieldBoosters() {
        return shieldBoosters;
    }

    public HangarToUI getHangar() {
        return hangar;
    }
    
    public DamageToUI getPendingDamage() {
        return pendingDamage;
    }
}
